pgjdbc/src/test/java/org/postgresql/test/core/v2/V2ParameterListTests.java

```diff
@@ -0,0 +1,94 @@
+/*-------------------------------------------------------------------------
+ *
+ * Copyright (c) 2003-2016, PostgreSQL Global Development Group
+ *
+ *
+ *-------------------------------------------------------------------------
+ */
+
+package org.postgresql.test.core.v2;
+
+import org.postgresql.core.ParameterList;
+
+import junit.framework.TestCase;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+
+/**
+ * Test cases to make sure the parameterlist implementation works as expected.
+ * Test case located in different package as the package private implementation.
+ * Reflection used to crack open and access the class. Not pretty but works.
+ *
+ * @author Jeremy Whiting jwhiting@redhat.com
+ *
+ */
+public class V2ParameterListTests extends TestCase {
+
+  public void testMergeOfParameterLists() {
+    try {
+      ClassLoader cl = this.getClass().getClassLoader();
+      Class cls = Class.forName("org.postgresql.core.v2.SimpleParameterList",
+          true, cl);
+      Constructor c = cls.getDeclaredConstructor(Integer.TYPE, Boolean.TYPE);
+      c.setAccessible(true);
+      Object o1SPL = c
+          .newInstance(new Object[] { new Integer(8), Boolean.TRUE });
+      assertNotNull(o1SPL);
+      Method msetIP = cls.getMethod("setIntParameter", Integer.TYPE,
+          Integer.TYPE);
+      assertNotNull(msetIP);
+      msetIP.setAccessible(true);
```

Comment:
> test classes package all start with the "test." 

I do not mind adding the test to the proper package (e.g. not `test.`) or even [Public Morozov](http://stackoverflow.com/a/6867140/1261287).

The problem with reflections is it is hard to maintain. "Find usages" does not work in IDE, "rename refactoring" does not work, etc, etc.

Here you propose a feature. It is not known how many users will actually use it (it is disabled by default), yet pgjdbc will have to maintain it. Tests is the only safety net, so having understandable test codebase is important in my view.